pygeoapi/provider/tinydb_.py

```diff
@@ -306,17 +314,69 @@ def delete(self, identifier):
 
         return True
 
-    def _bbox(input_bbox, record_bbox):
+    def _add_extra_fields(self, json_data: dict) -> dict:
         """
-        Test whether one bbox intersects another
+        Helper function to add extra fields to an item payload
 
-        :param input_bbox: `list` of minx,miny,maxx,maxy
-        :param record_bbox: `list` of minx,miny,maxx,maxy
+        :param json_data: `dict` of JSON data
 
-        :returns: `bool` of result
+        :returns: `dict` of updated JSON data
         """
 
-        return True
+        return json_data
+
+    def _add_search_query(self, query: list, search_term: str = None) -> str:
+        """
+        Helper function to add extra query predicates
+
+        :param query: `list` of query predicates
+        :param search_term: `str` of search term
+
+        :returns: `list` of updated query predicates
+        """
+
+        return query
+
+    def __repr__(self):
+        return f'<TinyDBProvider> {self.data}'
+
+
+class TinyDBCatalogueProvider(TinyDBProvider):
+    """TinyDB Catalogue Provider"""
+
+    def __init__(self, provider_def):
+        super().__init__(provider_def)
+
+        self._excludes = ['_metadata-anytext']
```

Comment:
Major: IMO will not have desired effect: `self.fields = self.get_fields()` is already called in the base class constructor. Or `self.fields = self.get_fields()` should be called again, but see my suggestion to make `_excludes` a class var as it is in effect a property of the classes.

Reply:
To consider: `self.fields` is never referenced after assignment so maybe `self.fields = self.get_fields()` can be removed and the above `self._excludes =` -statement will be ok. `get_fields()` is a Provider base.py method and the way to access fields I guess.

Reply:
If we remove `self.fields = self.get_fields()` from the constructor then this will affect other parts of `pygeoapi.api.itemtypes`.  Suggest we leave this as is in this PR and address in a subsequent PR later on.

Reply:
Ok, then the only remaining is IMO to resolve the proper `_excludes`.

Reply:
Do you agree that the current order will not work? Hence my suggestion for a class var (see my example) or another solution?

Reply:
The default TinyDB provider as a result of this PR has an empty `_excludes` member.  The TinyDBCatalogue provider sets `_excludes` to remove a given property that is a pygeoapi "extra".

Reply:
fwiw other providers make the call to `self.get_fields()` at the end of the init method which will either fetch self.fields if set, or set self.fields.

[esri.py](https://github.com/geopython/pygeoapi/blob/e8ce02167fd670bb515f8216d699a3e9abca21d4/pygeoapi/provider/esri.py#L70)

[sqlite.py](https://github.com/geopython/pygeoapi/blob/e8ce02167fd670bb515f8216d699a3e9abca21d4/pygeoapi/provider/sqlite.py#L82)

Happy to establish a normative behavior between the two peoples `self.fields = self.get_fields()` is present in

[postgresql.py](https://github.com/geopython/pygeoapi/blob/e8ce02167fd670bb515f8216d699a3e9abca21d4/pygeoapi/provider/postgresql.py#L127)

[elasticsearch_.py](https://github.com/geopython/pygeoapi/blob/e8ce02167fd670bb515f8216d699a3e9abca21d4/pygeoapi/provider/elasticsearch_.py#L90)

Reply:
Good points.  Suggest we address this as part of another PR